tests/cpp/test_layout_op.cpp

```diff
@@ -70,6 +70,70 @@ class LayoutOpTest : public NVFuserTest {
   }
 };
 
+TEST_F(LayoutOpTest, LogicalAndAllocationSizes) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  auto inp = makeSymbolicTensor(2);
+  fusion.addInput(inp);
+  auto out = set(inp);
+  fusion.addOutput(out);
+  // padding output to multiple of 16
+  out->split(1, 16);
+  out->setAllocationDomain(out->getLoopDomain(), true);
+  // restore loop domain
+  out->merge(1);
```

Comment:
This doesn't restore. Is this necessary?

Reply:
Touche. It unsplit the loop domain so that it has the same size as logical domain.
You are right that the extent is no longer the same, so it's not a *restoration*.

Schedulers expects un-scheduled fusion. Without this merge, I'm hitting the assert here:
https://github.com/NVIDIA/Fuser/blob/db9721d5d8591ca053f08ed7850ff66617daf0dd/csrc/scheduler/pointwise.cpp#L357

Reply:
I suppose you can just modify the allocation domain with AbstractTensor. I remember there are some tests.

Reply:
So, what did you decide to do? Nothing seems to have changed?

> I can also directly using IterDomain::split for that.

Of course, but you'd need to maintain the proper ordering of the ID vector yourself.